README.md
# OmegaT adaptive theme plugin

This is OmegaT theme plugin to provide adaptive coloring for OmegaT 5.6.0 and later.
The plugin allow omegat to select dark/light, high/normal contrast themes based on DarkLaf project.


## Build system

The plugin  use a Gradle build system as same as OmegaT version 4.3.0 and later.

## Dependency

OmegaT and dependencies are located on remote mavenCentral repositories.
It is necessary to connect the internet to compile your project.

The plugin refers OmegaT 5.7.1.

## License

The plugin is distributed under GNU General Public License 3 or later.



CHANGELOG.md
# Change Log
All notable changes to this project will be documented in this file.

## [Unreleased]

## [v0.1.2]

* Bump DarkLaf@3.0.0
* Move detection of user preferences out of constructor.

## [v0.1.1]

* Drop rounded buttons

## v0.1.0

* First public pre-release.

[Unreleased]: https://github.com/miurahr/omegat-adaptive-theme/compare/v0.1.2...HEAD
[v0.1.2]: https://github.com/miurahr/omegat-adaptive-theme/compare/v0.1.1...v0.1.2
[v0.1.1]: https://github.com/miurahr/omegat-adaptive-theme/compare/v0.1.0...v0.1.1


src/main/java/tokyo/northside/omegat/theme/AdaptiveTheme.java
/**************************************************************************
 OmegaT - Computer Assisted Translation (CAT) tool
          with fuzzy matching, translation memory, keyword search,
          glossaries, and translation leveraging into updated projects.

 Copyright (C) 2022 Hiroshi Miura
 Copyright (c) 2019-2022 Jannis Weis
               Home page: http://www.omegat.org/

 This file is part of OmegaT.

 OmegaT is free software: you can redistribute it and/or modify
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation, either version 3 of the License, or
 (at your option) any later version.

 OmegaT is distributed in the hope that it will be useful,
 but WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 GNU General Public License for more details.

 You should have received a copy of the GNU General Public License
 along with this program.  If not, see <http://www.gnu.org/licenses/>.
 **************************************************************************/
package tokyo.northside.omegat.theme;


import com.github.weisj.darklaf.DarkLaf;
import com.github.weisj.darklaf.LafManager;
import com.github.weisj.darklaf.theme.spec.ColorToneRule;
import com.github.weisj.darklaf.theme.spec.ContrastRule;
import com.github.weisj.darklaf.theme.spec.PreferredThemeStyle;
import org.omegat.gui.theme.DefaultFlatTheme;
import tokyo.northside.omegat.theme.impl.AdaptiveThemeManager;

import javax.swing.UIDefaults;
import javax.swing.UIManager;


/**
 * @author Hiroshi Miura
 */
public class AdaptiveTheme extends DarkLaf {
    private static final String NAME = "Adaptive Theme";
    private static final String DESCRIPTION = "Adaptive theme based on DarkLaf";

    /**
     * Default constructor.
     */
    public AdaptiveTheme() {
        super();
    }

    public static void loadPlugins() {
        UIManager.installLookAndFeel(NAME, AdaptiveTheme.class.getCanonicalName());
    }

    public static void unloadPlugins() {
    }

    public UIDefaults getDefaults() {
        // XXX: this does not detect dark mode on Linux
        PreferredThemeStyle style = LafManager.getPreferredThemeStyle();
        setTheme( LafManager.getThemeProvider().getTheme(style));
        UIDefaults defaults = super.getDefaults(); // get OmegaT defaults
        DefaultFlatTheme.setDefaults(defaults, getID());
        return AdaptiveThemeManager.getImplementation(isDark(), isHighContrast()).setDefaults(defaults);
    }

    private boolean isDark() {
        return getTheme().getColorToneRule() == ColorToneRule.DARK;
    }

    private boolean isHighContrast() {
        return getTheme().getContrastRule() == ContrastRule.HIGH_CONTRAST;
    }

    @Override
    public String getName() {
        return NAME;
    }

    @Override
    public String getDescription() {
        return DESCRIPTION;
    }

    @Override
    public boolean isNativeLookAndFeel() {
        return false;
    }

    @Override
    public boolean isSupportedLookAndFeel() {
        return true;
    }

}



src/main/java/tokyo/northside/omegat/theme/impl/AdaptiveThemeManager.java
package tokyo.northside.omegat.theme.impl;

import java.awt.Color;

public class AdaptiveThemeManager {

    /**
     * Adjust a color by adding some constant to its RGB values, clamping to the
     * range 0-255.
     */
    static Color adjustRGB(final Color color, final int adjustment) {
        return new Color(Math.max(0, Math.min(255, color.getRed() + adjustment)),
                Math.max(0, Math.min(255, color.getGreen() + adjustment)),
                Math.max(0, Math.min(255, color.getBlue() + adjustment)));
    }

    public static IAdaptiveThemeImpl getImplementation(final boolean dark, final boolean highContrast) {
        if (dark & highContrast) {
            return new DarkHighContrastAdaptiveThemeImpl();
        } else if (dark) {
            return new DarkStandardAdaptiveThemeImpl();
        } else if (highContrast) {
            return new LightHighContrastAdaptiveThemeImpl();
        } else {
            return new LightStandardAdaptiveThemeImpl();
        }
    }
}


src/main/java/tokyo/northside/omegat/theme/impl/DarkStandardAdaptiveThemeImpl.java
package tokyo.northside.omegat.theme.impl;

import javax.swing.UIDefaults;
import javax.swing.UIManager;
import javax.swing.border.MatteBorder;
import java.awt.Color;

public class DarkStandardAdaptiveThemeImpl implements IAdaptiveThemeImpl {
    @Override
    public UIDefaults setDefaults(UIDefaults defaults) {
            // set OmegaT color scheme for dark theme.
            Color standardBgColor = defaults.getColor("Panel.background");
            // Borders
            Color borderColor = new Color(0x8aa5b8);
            defaults.put("OmegaTBorder.color", borderColor);
            defaults.put("OmegaTDockablePanel.border", new MatteBorder(1, 1, 1, 1, borderColor));

            // Default application Colors
            defaults.put("OmegaT.activeSource", new Color(0x287128));
            defaults.put("OmegaT.source", new Color(0x82b682));
            defaults.put("OmegaT.noted", new Color(0xa0c0c0));
            defaults.put("OmegaT.untranslated", new Color(0x4d4daa));
            defaults.put("OmegaT.translated", new Color(0x57572d));
            defaults.put("OmegaT.nonUnique", new Color(0x808080));
            defaults.put("OmegaT.placeholder", new Color(0x969696));
            defaults.put("OmegaT.removeTextTarget", new Color(0xff0000));
            defaults.put("OmegaT.nbsp", new Color(0xc8c8c8));
            defaults.put("OmegaT.whiteSpace", new Color(0x808080));
            defaults.put("OmegaT.bidiMarkers", new Color(0x480000));
            defaults.put("OmegaT.paragraphStart", new Color(0xaeaeae));
            defaults.put("OmegaT.markComesFromTm", new Color(0xaa8072));
            defaults.put("OmegaT.markComesFromTxXice", new Color(0xaf76df));
            defaults.put("OmegaT.markComesFromTmX100pc", new Color(0x3f54f8));
            defaults.put("OmegaT.markComesFromTmXauto", new Color(0x3f2546));
            defaults.put("OmegaT.markComesFromTmXendorced", new Color(0x005500));
            defaults.put("OmegaT.replace", new Color(0x0000ff));
            defaults.put("OmegaT.languageTools", new Color(0x0000ff));
            defaults.put("OmegaT.transTips", new Color(0x0000ff));
            defaults.put("OmegaT.spellCheck", new Color(0xff0000));
            defaults.put("OmegaT.terminology", new Color(0xffa500));
            defaults.put("OmegaT.matchesChanged", new Color(0x0000ff));
            defaults.put("OmegaT.matchesUnchanged", new Color(0x00ff00));
            defaults.put("OmegaT.matchesInsActive", new Color(0x0000ff));
            defaults.put("OmegaT.matchesInsInactive", new Color(0x808080));
            defaults.put("OmegaT.hyperlink", new Color(0x0000ff));
            defaults.put("OmegaT.searchFoundMark", new Color(0x0000ff));
            defaults.put("OmegaT.searchReplaceMark", new Color(0xff9900));
            defaults.put("OmegaT.notificationMin", new Color(0x332233));
            defaults.put("OmegaT.notificationMax", new Color(0x647354));
            defaults.put("OmegaT.alignerAccepted", new Color(0x15bb45));
            defaults.put("OmegaT.alignerNeedsReview", new Color(0xff0000));
            defaults.put("OmegaT.alignerHighlight", new Color(0xffff00));
            defaults.put("OmegaT.alignerTableRowHighlight", new Color(0xc8c8c8));
            defaults.put("OmegaT.projectFilesCurrentFileForeground", new Color(0x0));
            defaults.put("OmegaT.projectFilesCurrentFileBackground", new Color(0xc8ddf2));
            defaults.put("OmegaT.searchFieldErrorText", new Color(0x7f0000));
            defaults.put("OmegaT.searchDimmedBackground", new Color(0x80, 0x80, 0x80, 0x80));
            defaults.put("OmegaT.searchResultBorder", new Color(0xEE, 0xD2, 0x00, 0x80));
            defaults.put("OmegaT.machinetranslateSelectedHighlight", new Color(0xff9900));

            // FIXME: VLDocking values have to be set to the "developer defaults"
            // not the "LAF defaults" because that's where
            // DockingUISettings#installUI puts them
            UIManager.put("DockViewTitleBar.border", new MatteBorder(1, 1, 1, 1, borderColor));

            // Panel title bars
            Color activeTitleBgColor = AdaptiveThemeManager.adjustRGB(standardBgColor, 0xF6 - 0xEE);
            Color activeTitleText = defaults.getColor("Label.foreground");
            Color inactiveTitleText = new Color(0x767676);

            defaults.put("InternalFrame.activeTitleForeground", activeTitleText);
            defaults.put("InternalFrame.activeTitleBackground", activeTitleBgColor);
            defaults.put("InternalFrame.inactiveTitleForeground", inactiveTitleText);
            defaults.put("InternalFrame.inactiveTitleBackground", standardBgColor);

            // Undocked panel
            defaults.put("activeCaption", Color.GRAY);
            defaults.put("activeCaptionBorder", borderColor);
            defaults.put("inactiveCaption", standardBgColor);
            defaults.put("inactiveCaptionBorder", borderColor);

            return defaults;
    }
}


src/main/java/tokyo/northside/omegat/theme/impl/LightStandardAdaptiveThemeImpl.java
package tokyo.northside.omegat.theme.impl;

import javax.swing.UIDefaults;
import javax.swing.UIManager;
import javax.swing.border.MatteBorder;
import java.awt.Color;

public class LightStandardAdaptiveThemeImpl implements IAdaptiveThemeImpl {

    public UIDefaults setDefaults(UIDefaults defaults) {
        // set OmegaT color scheme for light theme.
        Color standardBgColor = defaults.getColor("Panel.background");
        // Borders
        Color borderColor = new Color(0x0f5e9c);
        defaults.put("OmegaTBorder.color", borderColor);

        // OmegaT-defined Dockables.
        defaults.put("OmegaTDockablePanel.border", new MatteBorder(1, 1, 1, 1, borderColor));
        defaults.put("OmegaTDockablePanel.isProportionalMargins", true);

        // Default application Colors
        defaults.put("OmegaT.activeSource", new Color(0xc0ffc0));
        defaults.put("OmegaT.source", new Color(0xc0ffc0));
        defaults.put("OmegaT.noted", new Color(0xc0ffff));
        defaults.put("OmegaT.untranslated", new Color(0xc0c0ff));
        defaults.put("OmegaT.translated", new Color(0xffff99));
        defaults.put("OmegaT.nonUnique", new Color(0x808080));
        defaults.put("OmegaT.placeholder", new Color(0x969696));
        defaults.put("OmegaT.removeTextTarget", new Color(0xff0000));
        defaults.put("OmegaT.nbsp", new Color(0xc8c8c8));
        defaults.put("OmegaT.whiteSpace", new Color(0x808080));
        defaults.put("OmegaT.bidiMarkers", new Color(0xc80000));
        defaults.put("OmegaT.paragraphStart", new Color(0xaeaeae));
        defaults.put("OmegaT.markComesFromTm", new Color(0xfa8072));
        defaults.put("OmegaT.markComesFromTxXice", new Color(0xaf76df));
        defaults.put("OmegaT.markComesFromTmX100pc", new Color(0xff9408));
        defaults.put("OmegaT.markComesFromTmXauto", new Color(0xff9596));
        defaults.put("OmegaT.markComesFromTmXendorced", new Color(0xffccff));
        defaults.put("OmegaT.replace", new Color(0x0000ff));
        defaults.put("OmegaT.languageTools", new Color(0x0000ff));
        defaults.put("OmegaT.transTips", new Color(0x0000ff));
        defaults.put("OmegaT.spellCheck", new Color(0xff0000));
        defaults.put("OmegaT.terminology", new Color(0xff6600));
        defaults.put("OmegaT.matchesChanged", new Color(0x0000ff));
        defaults.put("OmegaT.matchesUnchanged", new Color(0x00ff00));
        defaults.put("OmegaT.matchesInsActive", new Color(0x0000ff));
        defaults.put("OmegaT.matchesInsInactive", new Color(0x808080));
        defaults.put("OmegaT.hyperlink", new Color(0x0000ff));
        defaults.put("OmegaT.searchFoundMark", new Color(0x0000ff));
        defaults.put("OmegaT.searchReplaceMark", new Color(0xff9900));
        defaults.put("OmegaT.notificationMin", new Color(0xfff2d4));
        defaults.put("OmegaT.notificationMax", new Color(0xff9900));
        defaults.put("OmegaT.alignerAccepted", new Color(0x15bb45));
        defaults.put("OmegaT.alignerNeedsReview", new Color(0xff0000));
        defaults.put("OmegaT.alignerHighlight", new Color(0xffff00));
        defaults.put("OmegaT.alignerTableRowHighlight", new Color(0xc8c8c8));
        defaults.put("OmegaT.projectFilesCurrentFileForeground", new Color(0x0));
        defaults.put("OmegaT.projectFilesCurrentFileBackground", new Color(0xc8ddf2));
        defaults.put("OmegaT.searchFieldErrorText", new Color(0xff0000));
        defaults.put("OmegaT.searchDimmedBackground", new Color(0x80, 0x80, 0x80, 0x80));
        defaults.put("OmegaT.searchResultBorder", new Color(0xEE, 0xD2, 0x00, 0x80));
        defaults.put("OmegaT.machinetranslateSelectedHighlight", new Color(0xffff00));

        // FIXME: VLDocking values have to be set to the "developer defaults"
        // not the "LAF defaults" because that's where
        // DockingUISettings#installUI puts them
        UIManager.put("DockViewTitleBar.border", new MatteBorder(1, 1, 1, 1, borderColor));

        int cornerRadius = 0;

        // Panel title bars
        Color activeTitleText = defaults.getColor("Label.foreground");
        Color activeTitleBgColor = AdaptiveThemeManager.adjustRGB(standardBgColor, 0xF6 - 0xED);
        Color inactiveTitleText = new Color(0x808080);
        defaults.put("InternalFrame.activeTitleForeground", activeTitleText);
        defaults.put("InternalFrame.activeTitleBackground", activeTitleBgColor);
        defaults.put("InternalFrame.inactiveTitleForeground", inactiveTitleText);
        defaults.put("InternalFrame.inactiveTitleBackground", standardBgColor);

        // Undocked panel
        defaults.put("activeCaption", Color.WHITE);
        defaults.put("activeCaptionBorder", borderColor);
        defaults.put("inactiveCaption", standardBgColor);
        defaults.put("inactiveCaptionBorder", borderColor);

        return defaults;
    }

}


src/main/java/tokyo/northside/omegat/theme/impl/LightHighContrastAdaptiveThemeImpl.java
package tokyo.northside.omegat.theme.impl;

import javax.swing.UIDefaults;
import javax.swing.UIManager;
import javax.swing.border.MatteBorder;
import java.awt.Color;

public class LightHighContrastAdaptiveThemeImpl implements IAdaptiveThemeImpl {

    public UIDefaults setDefaults(UIDefaults defaults) {
        // set OmegaT color scheme for light theme.
        Color standardBgColor = defaults.getColor("Panel.background");
        // Borders
        Color borderColor = new Color(0x0f5e9c);
        defaults.put("OmegaTBorder.color", borderColor);

        // OmegaT-defined Dockables.
        defaults.put("OmegaTDockablePanel.border", new MatteBorder(1, 1, 1, 1, borderColor));
        defaults.put("OmegaTDockablePanel.isProportionalMargins", true);

        // Default application Colors
        defaults.put("OmegaT.activeSource", new Color(0xc0ffc0));
        defaults.put("OmegaT.source", new Color(0xc0ffc0));
        defaults.put("OmegaT.noted", new Color(0xc0ffff));
        defaults.put("OmegaT.untranslated", new Color(0xc0c0ff));
        defaults.put("OmegaT.translated", new Color(0xffff99));
        defaults.put("OmegaT.nonUnique", new Color(0x808080));
        defaults.put("OmegaT.placeholder", new Color(0x969696));
        defaults.put("OmegaT.removeTextTarget", new Color(0xff0000));
        defaults.put("OmegaT.nbsp", new Color(0xc8c8c8));
        defaults.put("OmegaT.whiteSpace", new Color(0x808080));
        defaults.put("OmegaT.bidiMarkers", new Color(0xc80000));
        defaults.put("OmegaT.paragraphStart", new Color(0xaeaeae));
        defaults.put("OmegaT.markComesFromTm", new Color(0xfa8072));
        defaults.put("OmegaT.markComesFromTxXice", new Color(0xaf76df));
        defaults.put("OmegaT.markComesFromTmX100pc", new Color(0xff9408));
        defaults.put("OmegaT.markComesFromTmXauto", new Color(0xff9596));
        defaults.put("OmegaT.markComesFromTmXendorced", new Color(0xffccff));
        defaults.put("OmegaT.replace", new Color(0x0000ff));
        defaults.put("OmegaT.languageTools", new Color(0x0000ff));
        defaults.put("OmegaT.transTips", new Color(0x0000ff));
        defaults.put("OmegaT.spellCheck", new Color(0xff0000));
        defaults.put("OmegaT.terminology", new Color(0xff6600));
        defaults.put("OmegaT.matchesChanged", new Color(0x0000ff));
        defaults.put("OmegaT.matchesUnchanged", new Color(0x00ff00));
        defaults.put("OmegaT.matchesInsActive", new Color(0x0000ff));
        defaults.put("OmegaT.matchesInsInactive", new Color(0x808080));
        defaults.put("OmegaT.hyperlink", new Color(0x0000ff));
        defaults.put("OmegaT.searchFoundMark", new Color(0x0000ff));
        defaults.put("OmegaT.searchReplaceMark", new Color(0xff9900));
        defaults.put("OmegaT.notificationMin", new Color(0xfff2d4));
        defaults.put("OmegaT.notificationMax", new Color(0xff9900));
        defaults.put("OmegaT.alignerAccepted", new Color(0x15bb45));
        defaults.put("OmegaT.alignerNeedsReview", new Color(0xff0000));
        defaults.put("OmegaT.alignerHighlight", new Color(0xffff00));
        defaults.put("OmegaT.alignerTableRowHighlight", new Color(0xc8c8c8));
        defaults.put("OmegaT.projectFilesCurrentFileForeground", new Color(0x0));
        defaults.put("OmegaT.projectFilesCurrentFileBackground", new Color(0xc8ddf2));
        defaults.put("OmegaT.searchFieldErrorText", new Color(0xff0000));
        defaults.put("OmegaT.searchDimmedBackground", new Color(0x80, 0x80, 0x80, 0x80));
        defaults.put("OmegaT.searchResultBorder", new Color(0xEE, 0xD2, 0x00, 0x80));
        defaults.put("OmegaT.machinetranslateSelectedHighlight", new Color(0xffff00));

        // FIXME: VLDocking values have to be set to the "developer defaults"
        // not the "LAF defaults" because that's where
        // DockingUISettings#installUI puts them
        UIManager.put("DockViewTitleBar.border", new MatteBorder(1, 1, 1, 1, borderColor));

        int cornerRadius = 0;

        // Panel title bars
        Color activeTitleText = defaults.getColor("Label.foreground");
        Color activeTitleBgColor = AdaptiveThemeManager.adjustRGB(standardBgColor, 0xF6 - 0xED);
        Color inactiveTitleText = new Color(0x808080);
        defaults.put("InternalFrame.activeTitleForeground", activeTitleText);
        defaults.put("InternalFrame.activeTitleBackground", activeTitleBgColor);
        defaults.put("InternalFrame.inactiveTitleForeground", inactiveTitleText);
        defaults.put("InternalFrame.inactiveTitleBackground", standardBgColor);

        // Undocked panel
        defaults.put("activeCaption", Color.WHITE);
        defaults.put("activeCaptionBorder", borderColor);
        defaults.put("inactiveCaption", standardBgColor);
        defaults.put("inactiveCaptionBorder", borderColor);

        return defaults;
    }

}


src/main/java/tokyo/northside/omegat/theme/impl/DarkHighContrastAdaptiveThemeImpl.java
package tokyo.northside.omegat.theme.impl;

import javax.swing.UIDefaults;
import javax.swing.UIManager;
import javax.swing.border.MatteBorder;
import java.awt.Color;

public class DarkHighContrastAdaptiveThemeImpl implements IAdaptiveThemeImpl {
    @Override
    public UIDefaults setDefaults(UIDefaults defaults) {
            // set OmegaT color scheme for dark theme.
            Color standardBgColor = defaults.getColor("Panel.background");
            // Borders
            Color borderColor = new Color(0x8aa5b8);
            defaults.put("OmegaTBorder.color", borderColor);
            defaults.put("OmegaTDockablePanel.border", new MatteBorder(1, 1, 1, 1, borderColor));

            // Default application Colors
            defaults.put("OmegaT.activeSource", new Color(0x287128));
            defaults.put("OmegaT.source", new Color(0x82b682));
            defaults.put("OmegaT.noted", new Color(0xa0c0c0));
            defaults.put("OmegaT.untranslated", new Color(0x4d4daa));
            defaults.put("OmegaT.translated", new Color(0x57572d));
            defaults.put("OmegaT.nonUnique", new Color(0x808080));
            defaults.put("OmegaT.placeholder", new Color(0x969696));
            defaults.put("OmegaT.removeTextTarget", new Color(0xff0000));
            defaults.put("OmegaT.nbsp", new Color(0xc8c8c8));
            defaults.put("OmegaT.whiteSpace", new Color(0x808080));
            defaults.put("OmegaT.bidiMarkers", new Color(0x480000));
            defaults.put("OmegaT.paragraphStart", new Color(0xaeaeae));
            defaults.put("OmegaT.markComesFromTm", new Color(0xaa8072));
            defaults.put("OmegaT.markComesFromTxXice", new Color(0xaf76df));
            defaults.put("OmegaT.markComesFromTmX100pc", new Color(0x3f54f8));
            defaults.put("OmegaT.markComesFromTmXauto", new Color(0x3f2546));
            defaults.put("OmegaT.markComesFromTmXendorced", new Color(0x005500));
            defaults.put("OmegaT.replace", new Color(0x0000ff));
            defaults.put("OmegaT.languageTools", new Color(0x0000ff));
            defaults.put("OmegaT.transTips", new Color(0x0000ff));
            defaults.put("OmegaT.spellCheck", new Color(0xff0000));
            defaults.put("OmegaT.terminology", new Color(0xffa500));
            defaults.put("OmegaT.matchesChanged", new Color(0x0000ff));
            defaults.put("OmegaT.matchesUnchanged", new Color(0x00ff00));
            defaults.put("OmegaT.matchesInsActive", new Color(0x0000ff));
            defaults.put("OmegaT.matchesInsInactive", new Color(0x808080));
            defaults.put("OmegaT.hyperlink", new Color(0x0000ff));
            defaults.put("OmegaT.searchFoundMark", new Color(0x0000ff));
            defaults.put("OmegaT.searchReplaceMark", new Color(0xff9900));
            defaults.put("OmegaT.notificationMin", new Color(0x332233));
            defaults.put("OmegaT.notificationMax", new Color(0x647354));
            defaults.put("OmegaT.alignerAccepted", new Color(0x15bb45));
            defaults.put("OmegaT.alignerNeedsReview", new Color(0xff0000));
            defaults.put("OmegaT.alignerHighlight", new Color(0xffff00));
            defaults.put("OmegaT.alignerTableRowHighlight", new Color(0xc8c8c8));
            defaults.put("OmegaT.projectFilesCurrentFileForeground", new Color(0x0));
            defaults.put("OmegaT.projectFilesCurrentFileBackground", new Color(0xc8ddf2));
            defaults.put("OmegaT.searchFieldErrorText", new Color(0x7f0000));
            defaults.put("OmegaT.searchDimmedBackground", new Color(0x80, 0x80, 0x80, 0x80));
            defaults.put("OmegaT.searchResultBorder", new Color(0xEE, 0xD2, 0x00, 0x80));
            defaults.put("OmegaT.machinetranslateSelectedHighlight", new Color(0xff9900));

            // FIXME: VLDocking values have to be set to the "developer defaults"
            // not the "LAF defaults" because that's where
            // DockingUISettings#installUI puts them
            UIManager.put("DockViewTitleBar.border", new MatteBorder(1, 1, 1, 1, borderColor));

            // Panel title bars
            Color activeTitleBgColor = AdaptiveThemeManager.adjustRGB(standardBgColor, 0xF6 - 0xEE);
            Color activeTitleText = defaults.getColor("Label.foreground");
            Color inactiveTitleText = new Color(0x767676);

            defaults.put("InternalFrame.activeTitleForeground", activeTitleText);
            defaults.put("InternalFrame.activeTitleBackground", activeTitleBgColor);
            defaults.put("InternalFrame.inactiveTitleForeground", inactiveTitleText);
            defaults.put("InternalFrame.inactiveTitleBackground", standardBgColor);

            // Undocked panel
            defaults.put("activeCaption", Color.GRAY);
            defaults.put("activeCaptionBorder", borderColor);
            defaults.put("inactiveCaption", standardBgColor);
            defaults.put("inactiveCaptionBorder", borderColor);

            return defaults;
    }
}
